### PR TITLE
enhance serverless vs dedicated vs byoc

### PR DESCRIPTION
**Pull request by micheleRP**
## Description

Resolves https://github.com/redpanda-data/documentation-private/issues/2681 
Review deadline: Friday Aug 23

## Page previews
[Cloud Overview](https://deploy-preview-31--rp-cloud.netlify.app/redpanda-cloud/get-started/cloud-overview/)

### CODE REVIEW

File: modules/get-started/pages/cloud-overview.adoc [hunk 1]
```diff
@@ -7,71 +7,135 @@ Redpanda Cloud is a complete data streaming platform delivered as a fully-manage
...
-Redpanda offers three types of fully-managed cloud clusters:
+Redpanda offers three types of fully-managed cloud clusters: Serverless, Dedicated, and BYOC.
+
+|===
+| Serverless | Dedicated | Bring Your Own Cloud (BYOC)
+
+| Hosted in Redpanda Cloud | Hosted in Redpanda Cloud | Hosted in your private cloud
+| Instantly available with no base cost | Provides higher performance and reliability | Provides data sovereignty + greater security and isoloation
+| Cloud providers = AWS | Cloud providers = AWS, GCP | Cloud providers = AWS, GCP, Azure
+| Public networking | Private networking | Private networking
```
Comment (bpraseed) [inline modules/get-started/pages/cloud-overview.adoc:20]: Include Public networking for Dedicated and BYOC. It feels like we only support Private.

File: modules/get-started/pages/cloud-overview.adoc [hunk 2]
```diff
@@ -7,71 +7,135 @@ Redpanda Cloud is a complete data streaming platform delivered as a fully-manage
 
 TIP: For more detailed information about the Redpanda platform, see xref:get-started:intro-to-events.adoc[] and xref:get-started:architecture.adoc[].
 
-== Cluster types
+== Redpanda Cloud cluster types
 
-Redpanda offers three types of fully-managed cloud clusters:
+Redpanda offers three types of fully-managed cloud clusters: Serverless, Dedicated, and BYOC.
```
Comment (Feediver1) [inline modules/get-started/pages/cloud-overview.adoc:12]: As we expand or revise these offerings, may make more sense not to specify the number and instead say:
Redpanda offers the following fully-managed cloud cluster types:

File: modules/get-started/pages/cloud-overview.adoc [hunk 3]
```diff
@@ -7,71 +7,135 @@ Redpanda Cloud is a complete data streaming platform delivered as a fully-manage
...
-* <<Dedicated Cloud>>: Single-tenant clusters hosted in Redpanda Cloud. This provides more control over your deployment. 
-* <<Bring Your Own Cloud (BYOC)>>: Clusters hosted in your private cloud. This provides more control over your deployment and offers full data sovereignty. 
-+
 NOTE: With standard BYOC clusters, Redpanda manages security policies and resources for your VPC, including service accounts, subnetworks, IAM roles, firewall rules, and storage buckets. For the most security, you can manage these resources yourself with a xref:get-started:cluster-types/byoc/vpc-byo-gcp.adoc[customer-managed VPC].
 
-=== Serverless
+=== Serverless vs Dedicated vs BYOC
 
-With Serverless clusters, you host your data in Redpanda's VPC, and Redpanda handles automatic scaling, provisioning, operations, and maintenance. This is a production-ready deployment option with a cluster available instantly. There is no base cost, and with pay-as-you-go billing after the free trial, you only pay for what you consume. You can view detailed billing activity for each cluster and edit payment methods on the *Billing* page. 
+Serverless clusters are a good fit for the following:
```
Comment (Feediver1) [inline modules/get-started/pages/cloud-overview.adoc:36]:
```suggestion
Serverless clusters are a good fit for the following use cases:
```

File: modules/get-started/pages/cloud-overview.adoc [hunk 4]
```diff
@@ -7,71 +7,135 @@ Redpanda Cloud is a complete data streaming platform delivered as a fully-manage
...
+
+* Starter and growing workloads
+* Spiky workloads (that is, development environments, systems that only occasionally get busy, or workloads that come and go)
+* Fast and dynamic cluster creation: you can use a Serverless cluster as an isolated container for topics
+
+With Serverless (or if you sign up for Dedicated through the AWS Marketplace), you only pay for what you consume, without any commitment. A cluster is created instantly, so you can surface it in your applications (for example, for tenant isolation). If your workload increases, you can migrate it to a Dedicated or BYOC cluster.
```
Comment (Feediver1) [inline modules/get-started/pages/cloud-overview.adoc:42]:
```suggestion
With Serverless (and for Dedicated when procured through the AWS Marketplace), you only pay for what you consume, without any commitment. A cluster is created instantly, so you can surface it in your applications (for example, for tenant isolation). If your workload increases, you can migrate it to a Dedicated or BYOC cluster.
```

**Comment by micheleRP** [comment]
Closing this, as it's old, and we'll wait to update overview until we have more guidance about direction of Serverless and Dedicated offerings. Some edits from this PR were moved into https://github.com/redpanda-data/cloud-docs/pull/83.

**Pull request closed without merging**